### ISSUE TITLE
configure internet on hub and nodes behind company proxy

### ISSUE THREAD

**Issue by danish-monu** [SeleniumHQ/docker-selenium#622]
okay. things are going very fine with the standalone and debug versions of chrome and firefox containers. now i want to switch to hub and link the chrome and firefox node. but, i have a network issue.
the docker is itself running on suse server. i shared the nertwok between the standalone chrome container (without hub) with the "--network host" option while firing up the container. then i pass the proxy settings in the capabilities (such as this "capabilities.setCapability(CapabilityType.PROXY, proxy)" where proxy is localhost:port of suse). This bring internet to my chrome container.
NOW, i want to set up things similarly on hub and link chrome node with it. so, i again used the "--network host" for firing up the hub. then i link up chrome node to the hub. then, sent the selenium requests, and i observe that internet did not work.
i even tried to link chrome nodes to hub with "--network host" option. and the following was the error: "docker: Error response from daemon: conflicting options: host type networking can't be used with links. This would result in undefined behavior."

in short, please help me setup hub with chrome nodes on a suse machine which is behind the proxy - localhost:port.

thanks

**Comment by diemol** [comment]
Hi @danish-monu,

I think I might know what the issue is, but just to be sure, can you please paste here the commands you are using to start the containers?

**Comment by danish-monu** [comment]
THESE COMMANS WORK for the standalone chrome container (without selenium hub):
1. docker run --network host -d -p 4444:4444 selenium/standalone-chrome
2. capabilities.setCapability(CapabilityType.PROXY, proxy), where proxy is localhost:port

THESE COMMANDS DO NOT LEAD TO WORKING INTERNET CONNECTION in a chrome node managed by selenium hub:
1. docker run --network host -d -p 4444:4444 --name selenium-hub selenium/hub
AND
1. docker run --network host -d -p 4444:4444 --name selenium-hub selenium/hub
2. docker run --network host -d --link selenium-hub:hub selenium/node-chrome

**Comment by diemol** [comment]
Thanks for replying @danish-monu, I'll check this. It will probably involve some work that will lead to a release, it'll take some weeks.

**Comment by diemol** [comment]
Hi @danish-monu,

So we released a while ago some changes, and I tried locally your desired setup and I think it should work now. Here is what I tried:

* I am using a Ubuntu
* I have a proxy running locally, let's say at http://localhost:8888 
* I start the hub like this
```
docker run --network host -d -p 4444:4444 --name selenium-hub selenium/hub:3.8.1-dubnium
```
* I start the Chrome node like this:
```
docker run --network host -e HUB_HOST=localhost -v /dev/shm:/dev/shm selenium/node-chrome:3.8.1-dubnium
```
_Notice that I use `-e HUB_HOST=localhost`, since the hub is actually running on the host network_

* And finally I used in the test `capabilities.setCapability(CapabilityType.PROXY, proxy)`, and it worked.

Please give it a try and let us know how it worked.

**Comment by diemol** [comment]
@danish-monu I'll close this issue since I saw it working properly, if something didn't work, please feel free to reopen it and we can check it again.
